lib/src/piece/import.dart

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import '../back_end/code_writer.dart';
+import '../constants.dart';
+import 'piece.dart';
+
+/// An import or export directive.
+///
+/// Contains pieces for the keyword and URI, the optional `as` clause for
+/// imports, the configurations (`if` clauses), and combinators (`show` and
+/// `hide`).
+class ImportPiece extends Piece {
+  /// The main directive and its URI.
```

Comment:
[nit] How about

/// The directive keyword and it's URI.

I think of the entire thing as "the directive", and I don't know what "main directive" means